streamalert/scheduled_queries/state/state_manager.py

```diff
@@ -0,0 +1,109 @@
+"""
+Copyright 2017-present, Airbnb Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import json
+
+
+class StateManager:
+    """Encapsulation of a caching system that is currently backed by the filesystem"""
+
+    def __init__(self, logger=None):
+        self._logger = logger
+
+        self._data = {}
+
+    def set(self, key, value):
+        self._data[key] = value
+
+    def has(self, key):
+        return key in self._data
+
+    def get(self, key, fallback=None):
+        return self._data.get(key, fallback)
+
+    def delete(self, key):
+        del self._data[key]
+
+    @property
+    def keys(self):
+        return list(self._data.keys())
+
+    def _dangerously_set_all_data(self, data):
+        """
+        This method is NOT intended to be used by any classes outside of this module.
+        """
+        self._data = data
+
+    def _dangerously_get_all_data(self):
+        """
+        This method is NOT intended to be used by any classes outside of this module.
+        """
+        return self._data
+
+    @property
+    def data(self):
```

Comment:
Ah, this looks like the remnant of an old function that I don't use anymore, cuz I decided it would be better abstraction to use `_dangerously_get_all_data`